### DISCUSSION TITLE
Adding support for actionable adaptive cards in search results

### DISCUSSION THREAD

**Discussion by PaoloPia** [microsoft-search/pnp-modern-search#2146] [Ideas]
I've seen that you recently introduced support for Adaptive Cards rendering of search results. That's a very cool feature, thank you.
However, as far as I can see, the Adaptive Cards are not actionable.
So, I'm opening this issue to discuss a bit how we could add support for actionable cards.
One idea, that @wobba suggested, is to use the same approach that you already use with web components, in order to add a custom library handling the "onAction" event for the adaptive card instances.
What do you think? I will be more than happy to implement this feature as a PR, if you like.
Thanks.

**Comment by FranckyC** [comment]
Hi Paolo, that is right, we've implemented the support of Adaptive Cards in the `develop` branch to render results and it should be released soon.

By actionable cards, I guess you are referring to this: https://docs.microsoft.com/en-us/adaptive-cards/sdk/rendering-cards/javascript/actions? Currently, web components are supported inside adaptive cards through markdown (ex here: https://github.com/microsoft-search/pnp-modern-search/blob/develop/search-parts/src/layouts/results/simpleList/simple-list.json), so theorically, you could already achieve actionable behaviors. But adding a less "hacky" way to do that is definitely a great idea.

We could totally use the same extensibility approach as we are using for web components, layouts, sources ,etc.
If you already have the architecture in mind, feel free to add a PR so we can review it.

Thanks for the feedback!

**Comment by PaoloPia** [comment]
Hi @FranckyC,
here #2243 you can find my proposed solution. I already tested it on a real customer's project and it works as expected. Please, let me know if there is anything else that you want me to do/update on that PR. Thanks.

**Reply by wobba** [reply]
@PaoloPia please address the comments :)
